backend/src/main/java/com/bakdata/conquery/sql/conversion/cqelement/aggregation/DateAggregationCteStep.java

```diff
@@ -1,7 +1,89 @@
 package com.bakdata.conquery.sql.conversion.cqelement.aggregation;
 
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
 import com.bakdata.conquery.sql.conversion.model.CteStep;
+import com.bakdata.conquery.sql.conversion.model.NameGenerator;
+import com.bakdata.conquery.sql.conversion.model.QueryStep;
+import com.bakdata.conquery.sql.conversion.model.SqlTables;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+
+@Getter
+@RequiredArgsConstructor
+enum DateAggregationCteStep implements CteStep {
```

Comment:
I merged all the 3 date aggregation enum classes into 1, mainly because there was a lot of duplicated code and intersect/merge use the same tables.